openattribute.php

```diff
@@ -648,74 +657,119 @@ function openattribute_options_page() {
 	<?php
 }
 
+function openattribute_show_html_options() {
+    ?>
+    <div style="width:95%; padding:10px; border:1px solid black; margin-top:7px;"><b><u>HTML Options</u></b><br/>
+		<p>
+			This section allows you (if you wish) to style how the attribution will appear.
+		</p>
+		<p>Before<br/>
+		<textarea cols="100" name="openattribute_pre_license_html"><?php echo stripslashes( get_site_option( 'openattribute_pre_license_html' ) ); ?></textarea><br />
+		</p><p>After<br/>
+		<textarea cols="100" name="openattribute_post_license_html"><?php echo get_site_option( 'openattribute_post_license_html' ); ?></textarea><br />
+		   </p>
+	</div>
+	<?php
+}
+
+function openattribute_show_add_licenses_options() {
+    ?>
+    <h3><a name="licenses">Add Licenses</a></h3>
+	<p>Please enter the licenses you wish to use below. The format for the license is "<b>URL</b><b>,</b><b>text to display on screen</b>". If you wish to use a custom license, perhaps create a page on your blog to hold your licensing info. By default you have been provided with some <a href="http://www.creativecommons.org" target="_blank">Creative Commons licenses</a>.</p>
+	<textarea rows="5" cols="100" style="width:100%; height:200px;" name="openattribute_licenses" >
+	<?php
+
+	$string = get_site_option( 'openattribute_licenses' );
+
+	if ( $string == '' ) {
+		echo "http://creativecommons.org/licenses/by-nd/3.0,Attribution-NoDerivatives CC BY-ND\n";
+		echo "http://creativecommons.org/licenses/by-nc-sa/3.0,Attribution-NonCommercial-ShareAlike CC BY-NC-SA\n";
+		echo "http://creativecommons.org/licenses/by-sa/3.0,Attribution-ShareAlike CC BY-SA\n";
+		echo "http://creativecommons.org/licenses/by-nc/3.0,Attribution-NonCommercial CC BY-NC\n";
+		echo "http://creativecommons.org/licenses/by-nc-nd/3.0,Attribution-NonCommercial-NoDerivatives CC BY-NC-ND\n";
+	} else {
+		echo $string;
+	}
+	?>
+	</textarea>
+    <?php
+}
+
 function openattribute_postform() {
 	if ( ( isset( $_POST['submitted'] ) ) && ( $_POST['submitted'] == 'openattribute' ) ) {
-		update_option( 'openattribute_licenses', $_POST['openattribute_licenses'] );
+		if ( isset( $_POST['openattribute_licenses'] ) ) {
+            update_site_option( 'openattribute_licenses', $_POST['openattribute_licenses'] );
+        }
 
 		if ( isset( $_POST['openattribute_firstrun'] ) && 'on' === $_POST['openattribute_firstrun'] ) {
```

Comment:
Not too sure why, but this option never gets set. If you check the open attribute_firstrun checkbox and try to save it, the checkbox remains unchecked after save.

Reply:
Resolved